stdlib/sql/hdb.go

```diff
@@ -0,0 +1,211 @@
+		case time.Time:
+			// DATE, TIME types get scanned to time.Time by the driver too,
+			// but they have no counterpart in Flux therefore will be represented as string
+			switch m.sqlTypes[i].DatabaseTypeName() {
+			case "DATE":
+				row[i] = values.NewString(value.Format(layoutDate))
+			case "TIME":
+				row[i] = values.NewString(value.Format(layoutTime))
+			default:
+				row[i] = values.NewTime(values.ConvertTime(value))
+			}
+		case []uint8:
+			switch m.columnTypes[i] {
+			case flux.TFloat:
+				var out hdb.Decimal
+				err := out.Scan(value)
+				if err != nil {
+					return nil, err
+				}
+				newFloat, _ := (*big.Rat)(&out).Float64()
+				row[i] = values.NewFloat(newFloat)
+			default:
+				row[i] = values.NewString(string(value))
+			}
+		case nil:
+			row[i] = values.NewNull(flux.SemanticType(m.columnTypes[i]))
+		default:
+			execute.PanicUnknownType(flux.TInvalid)
+		}
+	}
+	return row, nil
+}
+
+func (m *HdbRowReader) InitColumnNames(names []string) {
+	m.columnNames = names
+}
+
+func (m *HdbRowReader) InitColumnTypes(types []*sql.ColumnType) {
+	fluxTypes := make([]flux.ColType, len(types))
+	for i := 0; i < len(types); i++ {
+		switch types[i].DatabaseTypeName() {
+		case "TINYINT", "SMALLINT", "INTEGER", "BIGINT":
+			fluxTypes[i] = flux.TInt
+		case "REAL", "DOUBLE", "DECIMAL":
+			fluxTypes[i] = flux.TFloat
+		case "TIMESTAMP": // not exactly correct (see Notes)
+			fluxTypes[i] = flux.TTime
+		default:
+			fluxTypes[i] = flux.TString
+		}
+	}
+	m.columnTypes = fluxTypes
+	m.sqlTypes = types
+}
+
+func (m *HdbRowReader) ColumnNames() []string {
+	return m.columnNames
+}
+
+func (m *HdbRowReader) ColumnTypes() []flux.ColType {
+	return m.columnTypes
+}
+
+func (m *HdbRowReader) SetColumnTypes(types []flux.ColType) {
+	m.columnTypes = types
+}
+
+func (m *HdbRowReader) SetColumns(i []interface{}) {
+	m.columns = i
+}
+
+func (m *HdbRowReader) Close() error {
+	if m.CloseFunc != nil {
+		return m.CloseFunc()
+	}
+	if err := m.Cursor.Err(); err != nil {
+		return err
+	}
+	return m.Cursor.Close()
+}
+
+func NewHdbRowReader(r *sql.Rows) (execute.RowReader, error) {
+	reader := &HdbRowReader{
+		Cursor: r,
+	}
+	cols, err := r.Columns()
+	if err != nil {
+		return nil, err
+	}
+	reader.InitColumnNames(cols)
+
+	types, err := r.ColumnTypes()
+	if err != nil {
+		return nil, err
+	}
+	reader.InitColumnTypes(types)
+
+	return reader, nil
+}
+
+var fluxToHdb = map[flux.ColType]string{
+	flux.TFloat:  "DOUBLE",
+	flux.TInt:    "BIGINT",
+	flux.TString: "NVARCHAR(5000)", // 5000 is the max
+	flux.TBool:   "BOOLEAN",
+	flux.TTime:   "TIMESTAMP", // not exactly correct (see Notes)
+}
+
+// HdbTranslateColumn translates flux colTypes into their corresponding SAP HANA column type
+func HdbColumnTranslateFunc() translationFunc {
+	return func(f flux.ColType, colName string) (string, error) {
+		s, found := fluxToHdb[f]
+		if !found {
+			return "", errors.Newf(codes.Internal, "SAP HANA does not support column type %s", f.String())
```

Comment:
I see. I'll fix it here. This can be found in `translationFunc` of all other databases as well.  I've added it to the list of fixes and improvements that should be done to `sql` package that I keep in bonitoo Flux fork https://github.com/bonitoo-io/flux/issues/1